### ISSUE TITLE
[Accordion Menu] Missing aria-selected state/attribute

### ISSUE THREAD

**Issue by ginestra** [foundation/foundation-sites#12205]
## Expected Behavior

The role="treeitem" elements, in this case the <li>s in the accordion menu, should have `aria-selected="true"` or false if the menu item is selected to follow WCAG 2.1 guidelines on aria states.
See https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html

## Current Behavior

<li> items don't have a valid ARIA state, failing A level WCAG 2.1

Actuall error:
Missing required WAI-ARIA states or properties
4.1.2 Name, Role, Value

## Possible Solution

Add conditional aria-selected attribute to the list items acting as treeitems.

## Test Case and/or Steps to Reproduce (for bugs)

The examples on the Foundation Docs fail to pass this too:
https://get.foundation/sites/docs/accordion-menu.html

How to reproduce:
1.
2.
3.

## Context

Foundation components are really useful as they come fully usable out of the box and they are mostly accessible.
In this case I had to override the flat_menu template to provide the extra attribute and pass the accessibility test.

## Your Environment

- Foundation version(s) used: Foundation Sites 6.4.3
- Browser(s) name and version(s): Firefox 85.0.2, Google Chrome 88.0.4324.192
- Device, Operating System and version: MacBook Pro, MacOS Big Sur 11.2
- Link to your project: https://cotr.ac.uk/

## Checklist

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] This is a bug report or a feature request.
- [x] There are no other issues similar to this one.
- [x] The issue title and template are correctly filled.

**Comment by angelicochris** [comment]
Yea site improve is bring this up as an issue for our site. Should it just be ignored as it doesn't apply to the accordion menu?
